rtloader/three/three.cpp

```diff
@@ -536,6 +537,38 @@ char **Three::getCheckWarnings(RtLoaderPyObject *check)
     return warnings;
 }
 
+char *Three::getCheckDiagnoses(RtLoaderPyObject *check)
+{
+    if (check == NULL) {
+        return NULL;
+    }
+
+    PyObject *py_check = reinterpret_cast<PyObject *>(check);
+
+    // result will be eventually returned as a copy and the corresponding Python
+    // string decref'ed, caller will be responsible for memory deallocation.
+    char *ret = NULL;
+    char func_name[] = "get_diagnoses";
+    PyObject *result = NULL;
+
+    result = PyObject_CallMethod(py_check, func_name, NULL);
+    if (result == NULL || !PyUnicode_Check(result)) {
+        setError("error invoking 'get_diagnoses' method: " + _fetchPythonError());
+        goto done;
+    }
+
+    ret = as_string(result);
+    if (ret == NULL) {
+        // as_string clears the error, so we can't fetch it here
+        setError("error converting 'get_diagnoses' result to string");
```

Comment:
Would it make sense to return the errors from this function to the caller (to be converted to a diagnosis struct perhaps) instead of setting it on the check? IIUIC this error message shows up on the agent status page, where it may be confusing since the check itself may be functioning normally.

Reply:
Good suggestion. Let me try to implement it. I was reluctant to do much reporting these kind of error transfer because I was followed existing patterns and was a bit afraid to deviate too much from them.